src/nvim/types.h

```diff
@@ -14,4 +14,20 @@ typedef unsigned char char_u;
 typedef uint32_t u8char_T;
 
 typedef struct expand expand_T;
+
+/// The scope of a working-directory command like `:cd`.
+///
+/// Scopes are enumerated from lowest to highest. When adding a scope make sure
+/// to update all functions using scopes as well, such as the implementation of
+/// `getcwd()`. When using scopes as limits (e.g. in loops) don't use the scopes
+/// directly, use `MIN_CD_SCOPE` and `MAX_CD_SCOPE` instead.
+typedef enum {
+  kCdScopeInvalid = -1,
+  kCdScopeWindow,  ///< Affects one window.
+  kCdScopeTab,     ///< Affects one tab page.
+  kCdScopeGlobal,  ///< Affects the entire instance of Neovim.
+} CdScope;
+#define MIN_CD_SCOPE  kCdScopeWindow
+#define MAX_CD_SCOPE  kCdScopeGlobal
```

Comment:
This should go in globals.h or vim.h, I think types.h only exists for unit tests and ultimately is something we want to eliminate. Its purpose was never clear.